src/subsystems/nodejs/translators/package-json/default.nix

```diff
@@ -40,26 +42,28 @@
       relPath=$(jq '.project.relPath' -c -r $jsonInput)
       npmArgs=$(jq '.project.subsystemInfo.npmArgs' -c -r $jsonInput)
 
+      # TODO: Do we really need to copy everything? Just package*.json is enough, no?
       cp -r $source/* ./
       chmod -R +w ./
       newSource=$(pwd)
 
       cd ./$relPath
-      rm -rf package-lock.json
 
-      if [ "$(jq '.project.subsystemInfo.noDev' -c -r $jsonInput)" == "true" ]; then
-        echo "excluding dev dependencies"
-        jq '.devDependencies = {}' ./package.json > package.json.mod
-        mv package.json.mod package.json
-        npm install --package-lock-only --production $npmArgs
-      else
+      # create package lock if missing or old
+      if ! [ -r package-lock.json ] || [ $(jq '.lockfileVersion' -r package-lock.json) != 2 ]; then
+        rm -f package-lock.json
+
+        # TODO: some easy way to add --offline, maybe separate flake run script? Speeds up enormously when cached.
         npm install --package-lock-only $npmArgs
       fi
 
+      # resolve packages - TODO move to RunCommandLocal
+      node ${./resolver.cjs} package-lock.json resolved.json
```

Comment:
@DavHau I tried moving the runcommandlocal to the pure translator but that doesn't get pkgs. Would it not be allowable to run pure commands in translators?

Like I said, it is way easier to resolve the packages in js than in nix, that cyclee inverter took me several hours where it would take me 10 minutes in js.